hazelcast/src/main/java/com/hazelcast/core/IMap.java

```diff
@@ -86,6 +86,13 @@
  * <li>Be careful while using default interface method implementations from {@link ConcurrentMap} and {@link Map}. Under
  * the hood they are typically implemented as a sequence of more primitive map operations, therefore the operations won't
  * be executed atomically.</li>
+ * <li>Currently, updates performed on the map are reflected in the indexes in
+ * a non-atomic way. Therefore, if there are indexes configured for the map,
+ * their state may slightly lag behind the state of the map. Use map listeners
+ * if you need to observe the state when the map and its indexes are consistent
+ * about the state of a particular map entry, see
+ * {@link #addEntryListener(MapListener, boolean) addEntryListener} for more
+ * details.</li>
```

Comment:
this explanation is valid for OBJECT and BINARY storage types i think. Is it also valid for NATIVE maps (since queries run on partition-threads)?

Reply:
You are right, the javadoc is updated.